### ISSUE TITLE
✨ [Feature] How to do graceful SHUTDOWN without main.go file

### ISSUE THREAD

**Issue by rajuaryan21** [goravel/goravel#436]
### Before feedback (在反馈之前)

- [X] I've searched the existing Issues, Discussions and Google (我已经搜索了现有的 Issues, Discussions 和 Google)
- [X] The problem can be stably reproduced (这个问题可以被稳定复现)
- [ ] The problem is generated after upgrading (问题是在升级之后产生的)

### Operating System (操作系统)

Linux

### Golang Version (Go 版本)

1.21.x

### Goravel Version (Goravel 版本)

1.14.x

### Describe The Problem (描述问题)

My question is pretty simple.

How do you gracefully shutdown your GO server without editing main.go file? is there any native way to do this? 

something like: signal.Notify(stopChannel, os.Interrupt) OR server.Shutdown(shutdownCtx)

Also, how do you perform some startup tasks before running server or creating htttp routes (all without editing main.go file)?

I am looking for a native way or at least some good practises inside goravel.

Thank you for replies. Goravel is an awesome framework.

### Reproduction Code (复现代码或截图)

```go
func shutdown() {
	fmt.Println("Shutdown signal received")
	defer func() {
		if err := recover(); err != nil {
			log.Printf("recovered: %+v\n", err)
		}
	}()
	d := time.Now().Add(5 * time.Second) // deadline 5s max
	ctx, cancel := context.WithDeadline(context.Background(), d)
	defer cancel()
	if err := srv.Shutdown(ctx); err != nil {
		log.Fatalf("could not shutdown: %v", err)
		os.Exit(1)
	}
}

func waitForSignal(s chan os.Signal, t chan struct{}) {
	select {
	case <-s:
		shutdown()
	case <-t:
		shutdown()
	}
}

--------------------------------------------

if verified {
 go func() {
    if err := facades.Route().Run(); err != nil {
       facades.Log().Errorf("Route run error: %v", err)
    }
 }()
}
```

**Comment by hwbrzzl** [comment]
Good idea, framework doesn't support it yet, but we can have it. I modified the type to feature, we can implement it in the future. For now, we need to modify the `main.go` file.

**Comment by rajuaryan21** [comment]
Just wondering how do I access the server instance to shutdown manually. Do you have any  example or something for me to get started?

**Comment by hwbrzzl** [comment]
`the server instance`?

**Comment by rajuaryan21** [comment]
yes

**Comment by rajuaryan21** [comment]
I want to give the option to stop the server at the click of a button for  SUPERADMIN. So, I need to access the server instance and shut it down. is it possible to access the server instance?

**Comment by hwbrzzl** [comment]
Okay, I got you. We can  add a method in v1.15, such as: `facades.Route().Shutdown()`. Thanks for your feedback!

**Comment by zxdstyle** [comment]
I also need this feature about `facades.Route().Shutdown()` very much, is there any release plan?

**Comment by hwbrzzl** [comment]
Hi @zxdstyle Thanks for your PRs. If you want to use this feature immediately, you can set the base branch of your PR to goravel/framework v1.14.x, goravel/gin v1.2.x and goravel/fiber v1.2.x. Otherwise, it will be released in v1.15, and this will take a long time.

**Comment by devhaozi** [comment]
I'm investigating the possibility of using `github.com/cloudflare/tableflip` and we need to export a new `Serve(l net.Listener) error` method.

**Comment by devhaozi** [comment]
Example: https://github.com/goravel/example/tree/tableflip
